src/main/java/org/opensearch/agent/tools/CreateAlertTool.java

```diff
@@ -0,0 +1,300 @@
+import static org.opensearch.ml.common.utils.StringUtils.isJson;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.commons.text.StringSubstitutor;
+import org.apache.logging.log4j.util.Strings;
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.admin.indices.get.GetIndexRequest;
+import org.opensearch.action.support.IndicesOptions;
+import org.opensearch.agent.tools.utils.ToolConstants.ModelType;
+import org.opensearch.agent.tools.utils.ToolHelper;
+import org.opensearch.client.Client;
+import org.opensearch.cluster.metadata.MappingMetadata;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.common.logging.LoggerMessageFormat;
+import org.opensearch.ml.common.FunctionName;
+import org.opensearch.ml.common.dataset.remote.RemoteInferenceInputDataSet;
+import org.opensearch.ml.common.input.MLInput;
+import org.opensearch.ml.common.output.model.ModelTensor;
+import org.opensearch.ml.common.output.model.ModelTensorOutput;
+import org.opensearch.ml.common.spi.tools.Tool;
+import org.opensearch.ml.common.spi.tools.ToolAnnotation;
+import org.opensearch.ml.common.transport.prediction.MLPredictionTaskAction;
+import org.opensearch.ml.common.transport.prediction.MLPredictionTaskRequest;
+import org.opensearch.ml.common.utils.StringUtils;
+
+import com.google.gson.reflect.TypeToken;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+@ToolAnnotation(CreateAlertTool.TYPE)
+public class CreateAlertTool implements Tool {
+    public static final String TYPE = "CreateAlertTool";
+
+    private static final String DEFAULT_DESCRIPTION =
+        "This is a tool that helps to create an alert(i.e. monitor with triggers), some parameters should be parsed based on user's question and context. The parameters should include: \n"
+            + "1. indices: The input indices of the monitor, should be a list of string in json format.\n";
+
+    @Setter
+    @Getter
+    private String name = TYPE;
+    @Getter
+    @Setter
+    private String description = DEFAULT_DESCRIPTION;
+
+    private final Client client;
+    private final String modelId;
+    private final String TOOL_PROMPT_TEMPLATE;
+
+    private static final String MODEL_ID = "model_id";
+    private static final String PROMPT_FILE_PATH = "CreateAlertDefaultPrompt.json";
+    private static final String DEFAULT_QUESTION = "Create an alert as your recommendation based on the context";
+
+    public CreateAlertTool(Client client, String modelId, String modelType) {
+        this.client = client;
+        this.modelId = modelId;
+        Map<String, String> promptDict = ToolHelper.loadDefaultPromptDictFromFile(this.getClass(), PROMPT_FILE_PATH);
```

Comment:
Can we make it static? Otherwise each time the agent executing, it will initialize the instance and load the json one more time.